code/edge/src/main/java/com/adobe/marketing/mobile/EdgeNetworkService.java

```diff
@@ -412,7 +412,18 @@ void handleStreamingResponse(
 
 		while (scanner.hasNext()) {
 			final String jsonResult = scanner.next();
-			responseCallback.onResponse(jsonResult.substring(trimLength));
+			try {
```

Comment:
Is it possible to avoid the exception all together by adding a check like this?
```
jsonResult.length > trimlength {
responseCallback.onResponse(jsonResult.substring(trimLength));
}
```